projects/plugins/jetpack/modules/sharedaddy/sharing-service.php

```diff
@@ -947,6 +947,12 @@ function sharing_display( $text = '', $echo = false ) {
 		return $text;
 	}
 
+	// Ensure we don't display sharing buttons on excerpts unless we are on search, archive or home page.
+	if ( ! is_search() && ! is_home() && ! is_archive() &&
```

Comment:
Should we have `is_front_page()` in there as well?

Reply:
@jeherve, thanks for pointing that out. I've checked it, and I think we shouldn't add a check for `is_front_page()`. When we set a home page to display the static page, it returns' true'. In such a case, it wouldn't remove sharing buttons from the Blog Posts block displayed on home.